internal/service/wallet.go

```diff
@@ -984,3 +999,267 @@ func (s *walletService) GetCustomerWallets(ctx context.Context, req *dto.GetCust
+	w, err := s.WalletRepo.GetWalletByID(ctx, walletID)
+	if err != nil {
+		return err
+	}
+
+	// Update alert state directly
+	w.AlertState = string(state)
+
+	return s.WalletRepo.UpdateWallet(ctx, walletID, w)
+}
+
+// PublishEvent publishes a webhook event for a wallet
+func (s *walletService) PublishEvent(ctx context.Context, eventName string, w *wallet.Wallet) error {
+	if s.WebhookPublisher == nil {
+		s.Logger.Warnw("webhook publisher not initialized", "event", eventName)
+		return nil
+	}
+
+	// Get real-time balance
+	balance, err := s.GetWalletBalance(ctx, w.ID)
+	if err != nil {
+		s.Logger.Errorw("failed to get wallet balance for webhook",
+			"wallet_id", w.ID,
+			"error", err,
+		)
+		return err
+	}
+
+	// Create internal event
+	internalEvent := &webhookDto.InternalWalletEvent{
+		EventType: eventName,
+		WalletID:  w.ID,
+		TenantID:  w.TenantID,
+	}
+
+	// Add alert info for alert events
+	if w.AlertConfig != nil && w.AlertConfig.Threshold != nil {
+		currentBalance := balance.RealTimeBalance
+		if currentBalance == nil {
+			currentBalance = &w.Balance
+		}
+		creditBalance := balance.RealTimeCreditBalance
+		if creditBalance == nil {
+			creditBalance = &w.CreditBalance
+		}
+
+		internalEvent.Alert = &webhookDto.WalletAlertInfo{
+			State:          w.AlertState,
+			Threshold:      w.AlertConfig.Threshold.Value,
+			CurrentBalance: *currentBalance,
+			CreditBalance:  *creditBalance,
+			AlertConfig:    w.AlertConfig,
+			AlertType:      getAlertType(eventName),
+		}
+	}
+
+	// Convert to JSON
+	eventJSON, err := json.Marshal(internalEvent)
+	if err != nil {
+		return ierr.WithError(err).
+			WithHint("Failed to marshal internal event").
+			Mark(ierr.ErrInternal)
+	}
+
+	webhookEvent := &types.WebhookEvent{
+		ID:            types.GenerateUUID(),
+		EventName:     eventName,
+		TenantID:      w.TenantID,
+		EnvironmentID: w.EnvironmentID,
+		UserID:        types.GetUserID(ctx),
+		Timestamp:     time.Now().UTC(),
+		Payload:       eventJSON,
+	}
+
+	s.Logger.Infow("publishing webhook event",
+		"event_id", webhookEvent.ID,
+		"event_name", eventName,
+		"wallet_id", w.ID,
+		"alert_state", w.AlertState,
+		"alert_config", w.AlertConfig,
+	)
+
+	return s.WebhookPublisher.PublishWebhook(ctx, webhookEvent)
+}
+
+func getAlertType(eventName string) string {
+	switch eventName {
+	case types.WebhookEventWalletCreditBalanceDropped:
+		return "credit_balance"
+	case types.WebhookEventWalletOngoingBalanceDropped:
+		return "ongoing_balance"
+	default:
+		return ""
+	}
+}
+
+// CheckBalanceThresholds checks if wallet balance is below threshold and triggers alerts
+func (s *walletService) CheckBalanceThresholds(ctx context.Context, w *wallet.Wallet, balance *dto.WalletBalanceResponse) error {
+	// Skip if alerts not enabled or no config
+	if !w.AlertEnabled || w.AlertConfig == nil || w.AlertConfig.Threshold == nil {
+		return nil
+	}
```

Comment:
there should be some option to store the alert at a tenant level as well and if it is enabled and the threshold can be defined at a tenant level and takes precedence if customer level alert config is missing,